[CI] StatelessSnapshotStressTestsIT testRandomActivitiesStatelessSnapshotDisabled failing

**Build Scans:**
- [elasticsearch-serverless-validate-submodule #23077 / java_run_serverless_elasticsearch_tests](https://gradle-enterprise.elastic.co/s/ksv662uol4pi6)
- [elasticsearch-serverless-es-pr-check #76552 / java_checkpart1](https://gradle-enterprise.elastic.co/s/i7hgcryvfpvmm)
- [elasticsearch-serverless-es-pr-check #75670 / java_checkpart1](https://gradle-enterprise.elastic.co/s/fxdhdltuc2alq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotStressTestsIT.testRandomActivitiesStatelessSnapshotDisabled" -Dtests.seed=B413832535724918 -Dtests.locale=ebu -Dtests.timezone=Brazil/DeNoronha -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotStressTestsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRandomActivitiesStatelessSnapshotDisabled'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: failed to acquire all permits: [index-0, node_t2, node_t1, node_t0]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testRandomActivitiesStatelessSnapshotDisabled
- [main] 1 consecutive failures in step java_run_serverless_elasticsearch_tests
- [main] 1 consecutive failures in pipeline elasticsearch-serverless-validate-submodule
- [main] 3 failures in test testRandomActivitiesStatelessSnapshotDisabled (0.4% fail rate in 774 executions)
- [main] 2 failures in step java_checkpart1 (0.3% fail rate in 647 executions)
- [main] 2 failures in pipeline elasticsearch-serverless-es-pr-check (0.3% fail rate in 596 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

---

Migrated from https://github.com/elastic/elasticsearch-serverless/issues/6315

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -418,9 +418,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.objectstore.AzureMultiProjectObjectStoreIT
   method: testObjectStoreSeparation
   issue: https://github.com/elastic/elasticsearch-serverless/issues/6311
-- class: org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotStressTestsIT
-  method: testRandomActivitiesStatelessSnapshotDisabled
-  issue: https://github.com/elastic/elasticsearch/issues/147845
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardDisruptionIT
   method: testReshardWithDisruption
   issue: https://github.com/elastic/elasticsearch/issues/147846
@@ -562,9 +559,6 @@ tests:
 - class: org.elasticsearch.search.CCSDuelIT
   method: testCardinalityAgg
   issue: https://github.com/elastic/elasticsearch/issues/148091
-- class: org.elasticsearch.xpack.stateless.snapshots.StatelessSnapshotStressTestsIT
-  method: testRandomActivitiesStatelessSnapshotReadFromObjectStore
-  issue: https://github.com/elastic/elasticsearch/issues/148092
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=mtermvectors/10_basic/Tests catching other exceptions per item}
   issue: https://github.com/elastic/elasticsearch/issues/148096

